Content.Server/Implants/ChameleonControllerSystem.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Content.Server.Clothing.Systems;
+using Content.Server.Preferences.Managers;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Preferences;
+using Content.Shared.Preferences.Loadouts;
+using Content.Shared.Roles;
+using Content.Shared.Station;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Implants;
+
+public sealed class ChameleonControllerSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+    [Dependency] private readonly SharedStationSpawningSystem _stationSpawningSystem = default!;
+    [Dependency] private readonly ChameleonClothingSystem _chameleonClothingSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<SubdermalImplantComponent, ChameleonControllerSelectedJobMessage>(OnSelected);
+    }
+
+    private void OnSelected(EntityUid uid, SubdermalImplantComponent component, ChameleonControllerSelectedJobMessage args)
+    {
+        if (component.ImplantedEntity == null || !HasComp<ChameleonControllerImplantComponent>(uid))
+            return;
+
+        ChangeChameleonClothingToJob(component.ImplantedEntity.Value, args.SelectedJob);
+    }
+
+    /// <summary>
+    ///     Switches all the chamelean clothing that the implant user is wearing to look like the selected job.
+    /// </summary>
+       private void ChangeChameleonClothingToJob(EntityUid user, ProtoId<JobPrototype> job)
+    {
+        if (!_proto.TryIndex(job, out var jobPrototype))
+            return;
+
+        if (!_proto.TryIndex(jobPrototype.StartingGear, out var startingGearPrototype))
+            return;
+
+        if (!TryComp<ActorComponent>(user, out var actorComponent))
+            return;
+
+        var session = actorComponent.PlayerSession;
+        var userId = actorComponent.PlayerSession.UserId;
+        var preferencesManager = IoCManager.Resolve<IServerPreferencesManager>();
+        var prefs = preferencesManager.GetPreferences(userId);
+
+        if (prefs.SelectedCharacter is not HumanoidCharacterProfile profile)
+            return;
+
+        if (!_inventorySystem.TryGetSlots(user, out var slots))
+            return;
+
+        // Does the job even exist?
+        var jobProtoId = "Job"+job.Id;
```

Comment:
You should be using 

```suggestion
var jobProtoId = LoadoutSystem.GetJobPrototype(job.Id);
```

Instead of hard coding this.

Example:

https://github.com/space-wizards/space-station-14/blob/782a2978f0b34f9c74b209a072a175e84e645b24/Content.Server/Administration/Commands/SetOutfitCommand.cs#L144-L146

Reply:
I was trying to get it to work but the static function was being annoying and I forgot to finish figuring it out